core/bin/dust_api.rs

```diff
@@ -50,16 +52,22 @@ struct RunManager {
 
 struct APIState {
     store: Box<dyn store::Store + Sync + Send>,
+    databases_store: Box<dyn databases_store::DatabasesStore + Sync + Send>,
     qdrant_clients: QdrantClients,
 
     run_manager: Arc<Mutex<RunManager>>,
 }
 
 impl APIState {
-    fn new(store: Box<dyn store::Store + Sync + Send>, qdrant_clients: QdrantClients) -> Self {
+    fn new(
+        store: Box<dyn store::Store + Sync + Send>,
+        databases_store: Box<dyn databases_store::DatabasesStore + Sync + Send>,
```

Comment:
I think this might be controversial @spolu 

The rationale is that we no longer have the ability to have a single worker be owner of a Table (because tables can now be in several DBs), so it doesn't really make sense to forward writes to the `databases_store` to a worker. I started doing it, but it involved picking a random sqlite worker etc.. It makes a lot more sense to just have `core` write directly to the DB. It allows to delete a good amount of pedestrian plumbing code as well.